src/ledger_api.rs

```diff
@@ -0,0 +1,123 @@
+use ledger::*;
+
+use super::*;
+
+use helium_api::Client;
+
+use super::traits::B58;
+use crate::result::Result;
+use keypair::PubKeyBin;
+use prettytable::Table;
+
+const INS_GET_PUBLIC_KEY: u8 = 0x02;
+const INS_SIGN_PAYMENT_TXN: u8 = 0x08;
+
+enum PubkeyDisplay {
+    Off,
+    On,
+}
+
+fn exchange_tx_get_pubkey(ledger: &LedgerApp, display: PubkeyDisplay) -> Result<PubKeyBin> {
+    let get_public_key = ApduCommand {
+        cla: 0xe0,
+        ins: INS_GET_PUBLIC_KEY,
+        p1: match display {
+            PubkeyDisplay::On => 0x01,
+            PubkeyDisplay::Off => 0x00,
+        },
+        p2: 0x00,
+        length: 0,
+        data: Vec::new(),
+    };
+
+    let public_key_result = ledger.exchange(get_public_key)?;
+    // TODO: verify validity before returning by checking the sha256 checksum
+    Ok(PubKeyBin::from_vec(&public_key_result.data[1..34]))
+}
+
+pub fn load_wallet() -> Result {
+    let ledger = LedgerApp::new()?;
+    let keypair = exchange_tx_get_pubkey(&ledger, PubkeyDisplay::On)?;
+
+    let mut table = Table::new();
+    table.add_row(row!["Address", "Type"]);
+    table.add_row(row![keypair.to_b58()?, "Ledger"]);
+    table.printstd();
+
+    Ok(())
+}
+
+fn transform_u64_to_array_of_u8(x: u64) -> [u8; 8] {
```

Comment:
(Louis and I already spoke offline about deleting this function and using `ByteOrder::write_u64::<LittleEndian>()` instead.)